### DISCUSSION TITLE
Custom Authentication with TikTok - Social login

### DISCUSSION THREAD

**Discussion by joshisar** [Web3Auth/Web3Auth#620] [Help]
Hi!

We wanted to integrate Login-with-Tok-Tok flow in our application using Web3Auth.Needed help to integrate the same using Custom Authentication.

From TikTok's dev document, we do get an open_id token in JWT format from the access-token endpoint. Regarding the JWKS uri, I have dropped a mail to TikTok's support asking the same as I couldn't find it in their docs.

Once we have that JWKS uri, how can we proceed further ?

Thanks & Regards,
Sarthak

**Comment by rinzler99** [comment]
You should get id_token and have jwks uri, once you get id_token and have jwks uri you can go to developer dashboard to setup a custom verifier. See below screenshot for the reference:
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/30899350/191003444-0f590db9-e87f-4088-a8b9-a401942bf6ff.png">

After setting up the custom verifier refer below link to integrate with web3auth:
https://web3auth.io/docs/sdk/web/core/custom-authentication

**Reply by joshisar** [reply]
Ok, so if TikTok doesn't do JWT token_id and openID, can we do this social login via AuthO using TikTok's OAuth2.0 servers in Web3Auth?

So we plan to connect to TikTok Auth servers via AuthO by creating a New Custom Social Connection for TikTok using this guide -> https://auth0.com/docs/authenticate/identity-providers/social-identity-providers/oauth2.


Then we add this AuthO option in web3auth. And then AuthO returns a token_id hopefully and keys are generated  ? Is this feasible ?
 
Thanks ,
Sarthak

**Reply by joshisar** [reply]
Also, I found an AuthO example which has Tok Tok login via AuthO in your GitHub repo : https://github.com/Web3Auth/examples/tree/main/web-core-sdk/custom-authentication/auth0-react-core-example

But it seems it's broken.

**Reply by irux** [reply]
did you look at this ? https://github.com/Web3Auth/Web3Auth/tree/master/demo/auth0-react-app

**Comment by rinzler99** [comment]
Hello Sarthak, 
You are correct, we can use auth0 and web3auth example for tik tok login. I guess you have already completed the tiktok and auth0 integration now you just need a simple guide on "how to connect auth0 with web3auth". @irux shared the [link](https://github.com/Web3Auth/Web3Auth/tree/master/demo/auth0-react-app) with you for the demo pls take a look at it. 
And also we have a look at the guide on docs page for setting up auth0 integration with web3auth. Please find below link.
https://web3auth.io/docs/guides/auth0

**Reply by rinzler99** [reply]
This would resolve your issue. Lmk if you have any doubts.

**Reply by joshisar** [reply]
@rinzler99 @shahbaz17  Connecting auth0 with web3auth is straightforward from the docs. Thanks for the link. But I am actually having trouble in connecting Auth0 to TikTok.

![Screenshot 2022-09-23 at 5 27 27 PM](https://user-images.githubusercontent.com/99479111/191958079-9de873fe-e8ad-4bcd-891f-7472f76e7989.jpg)

![Screenshot 2022-09-23 at 5 26 47 PM](https://user-images.githubusercontent.com/99479111/191956645-14395e1a-77ea-4b62-bace-54b32491489b.jpg)

My client_id is correct. Is it because my app is not approved in Tik Tok dev dashboard ? I have provided them a correct website URL but the app still isn't approved for production. Have contacted their support. Have you faced similar issue ?

**Reply by shahbaz17** [reply]
Hey @joshisar 
<img width="448" alt="Screenshot 2022-09-24 at 1 52 04 AM" src="https://user-images.githubusercontent.com/6962565/192027741-320ef6d1-c5cf-4fa5-8827-56e46a605d6c.png">

Simply append the `client_key` like this. TikTok needs client_key but the Auth0 dashboard has `Client ID`.

Also, can you share your masked Authorization and Token URL here?
